app/commands/v2/publish.rb

```diff
@@ -131,10 +130,10 @@ def publish_redirect_if_content_item_has_moved(new_location, previous_location,
         ) if draft_redirect
       end
 
-      def lookup_published_item(content_item)
+      def lookup_previous_item(content_item)
         ContentItemFilter.similar_to(
           content_item,
-          state: "published",
+          state: %w(published unpublished),
           base_path: nil,
           user_version: nil,
         ).first
```

Comment:
I'm with @tuzz on this one, in that it's best to provide a consistent worldview in which to work and then work more simply within it, rather than constantly validating that worldview whenever taking any action.

Up to him whether he implements the check, though.